src/engine/ExportQueryExecutionTrees.cpp

```diff
@@ -700,22 +700,22 @@ ExportQueryExecutionTrees::getLiteralOrNullopt(
 // _____________________________________________________________________________
 std::optional<LiteralOrIri>
 ExportQueryExecutionTrees::idToLiteralOrIriForEncodedValue(Id id) {
-  auto idLiteralAndType = idToStringAndTypeForEncodedValue(id);
-  if (idLiteralAndType.has_value()) {
-    auto lit = ad_utility::triple_component::Literal::literalWithoutQuotes(
-        idLiteralAndType.value().first);
-    lit.addDatatype(
-        ad_utility::triple_component::Iri::fromIrirefWithoutBrackets(
-            idLiteralAndType.value().second));
-    return LiteralOrIri{lit};
+  auto [literal, type] = idToStringAndTypeForEncodedValue(id).value_or(
```

Comment:
1. Add a TODO that this is still somewhat fishy and/or broken:

* `nullptr` happens for blank nodes and EncodedIRIs (which should be counted as IRIs).
* For `EncodedIRIs` this is typically no problem, because (other than suggested by the name) the `idToLiteralOrIri` function which is typically called handles `EncodedValue` before calling this function.
* For blank nodes, the nullptr -> nullopt is correct because blank nodes are neither literals nor irirs, but handling this via a nullptr is still dangerous and hard to reason about.